src/elements/Gallery/Gallery.tsx

```diff
@@ -76,4 +75,33 @@ export const Gallery: FunctionComponent<Props> = ({
             </Lightbox>
         </figure>
     );
-};
+}
+
+function Row(props: { tiles: Tile[]; margin: number; onClick: (image: UploadcareImage) => void }) {
+    const { margin, onClick, tiles } = props;
+
+    if (tiles.length === 0) {
+        return null;
+    }
+
+    const tilesWidth = tiles.map(({ width }) => width).reduce((sum, width) => sum + width, 0);
+
+    return (
+        <div className="prezly-slate-gallery__row">
+            {tiles.map(({ height, image, width }) => (
+                <GalleryImage
+                    key={image.uuid}
+                    image={image}
+                    onClick={onClick}
+                    rounded={margin > 0}
+                    style={{
+                        flexBasis: `${(100 * width) / tilesWidth}%`,
+                        width: `${(100 * width) / tilesWidth}%`,
+                        height: `${(100 * height) / tilesWidth}%`,
+                        margin: margin,
```

Comment:
This change make the gallery layout relative to width, so SSR-produced result will nicely scale to any width available in real browser.